compiler/ballerina-lang/src/main/java/io/ballerina/projects/BalaWriter.java

```diff
@@ -157,6 +157,13 @@ private void addPackageJson(ZipOutputStream balaOutputStream, Optional<JsonArray
         if (!platformLibs.isEmpty()) {
             packageJson.setPlatformDependencies(platformLibs.get());
         }
+
+        // Set icon in bala path ion the package.json
+        if (packageManifest.icon() != null && !packageManifest.icon().isEmpty()) {
+            Path iconPath = getIconPath(packageManifest.icon());
+            packageJson.setIcon(String.valueOf(Paths.get("docs").resolve(iconPath.getFileName())));
```

Comment:
We have a constant for `docs` dir in ProjectConstants

Reply:
Used it